src/Compilers/CSharp/Portable/Symbols/SymbolExtensions.cs

```diff
@@ -832,5 +819,16 @@ internal static int GetOverloadResolutionPriority(this Symbol symbol)
             Debug.Assert(symbol is MethodSymbol or PropertySymbol);
             return symbol is MethodSymbol method ? method.OverloadResolutionPriority : ((PropertySymbol)symbol).OverloadResolutionPriority;
         }
+
+        internal static bool IsExtensionParameter(this ParameterSymbol parameter)
+        {
+            return parameter.ContainingSymbol is TypeSymbol { IsExtension: true };
+        }
+
+        internal static bool IsExtensionParameterImplementation(this ParameterSymbol parameter)
+        {
+            return parameter.ContainingSymbol is SourceExtensionImplementationMethodSymbol implementationMethod
+                && parameter.Ordinal == 0;
```

Comment:
>parameter.Ordinal == 0 [](http://example.com/codeflow?start=19&length=22)

I think this condition is in insufficient and will lead to an incorrect result for static extension methods